apps/emails-and-messages/src/pages/api/webhooks/notify.ts

```diff
@@ -0,0 +1,123 @@
+import { NextWebhookApiHandler, SaleorAsyncWebhook } from "@saleor/app-sdk/handlers/next";
+import { saleorApp } from "../../../saleor-app";
+import { logger as pinoLogger } from "../../../lib/logger";
+import { sendEventMessages } from "../../../modules/event-handlers/send-event-messages";
+import { createClient } from "../../../lib/create-graphql-client";
+import { MessageEventTypes } from "../../../modules/event-handlers/message-event-types";
+
+// Notify event handles multiple event types which are recognized based on payload field `notify_event`.
+// Handler recognizes if event is one of the supported typed and sends appropriate message.
+
+export interface NotifyPayload {
+  notify_event: string;
+  payload: EventPayload;
+  meta: Meta;
+}
+
+export interface Meta {
+  issued_at: Date;
+  version: string;
+  issuing_principal: IssuingPrincipal;
+}
+
+export interface IssuingPrincipal {
+  id: null | string;
+  type: null | string;
+}
+
+export interface EventPayload {
+  user: User;
+  recipient_email: string;
+  channel_slug: string;
+  domain: string;
+  site_name: string;
+  logo_url: string;
+  token?: string;
+  confirm_url?: string;
+  reset_url?: string;
+  delete_url?: string;
+  old_email?: string;
+  new_email?: string;
+  redirect_url?: string;
+}
+
+export interface User {
+  id: string;
+  email: string;
+  first_name: string;
+  last_name: string;
+  is_staff: boolean;
+  is_active: boolean;
+  private_metadata: Metadata;
+  metadata: Metadata;
+  language_code: string;
+}
+
+export interface Metadata {}
```

Comment:
Nitpick: does all of that must be exported? I don't see all of them being used outside of this file and they have quite generic names (e.g. `User` or `Meta`) which may get messed up when trying to import something else.

Reply:
You are right. Interfaces no longer exposed.